### ISSUE TITLE
DAV\Client: Request body missing on PROPFIND after HEAD request

### ISSUE THREAD

**Issue by cweiske** [fruux/sabre-dav#649]
When using the `Sabre\DAV\Client` to do a `HEAD` request, followed by a `PROPFIND` request, the latter fails because of a missing response body.

Reason for this problem is that the curl settings (including the `CURLOPT_NOBODY` option) are not cleared/reset between requests.

Test case:
```
<?php
require 'vendor/autoload.php';
use Sabre\DAV;

$client = new Sabre\DAV\Client(
    array(
        'userName' => 'admin',
        'password' => 'admin',
        'baseUri'  => 'http://wolke.bogo/remote.php/webdav/'
    )
);
//check if file exists
$client->request('HEAD', 'Notes');
//then fetch properties
$res = $client->propFind('Notes', array('{DAV:}resourcetype'), 1);
var_dump($res);
?>
```

Result:
```
PHP Fatal error:  Uncaught exception 'InvalidArgumentException' with message
 'The body passed to parseMultiStatus could not be parsed. Is it really xml?'
```

This happens with `sabre/dav 2.1.3`.

**Comment by andreaswolf** [comment]
I also experienced this. I could fix it by recreating the cURL instance, but that is probably not the preferred way to do this.

**Comment by cweiske** [comment]
@andreaswolf: You once fixed it by always setting the `CURLOPT_NOBODY` option; see https://github.com/andreaswolf/SabreDAV/commit/aefe0c932ded46c0b612e330c875df08377e6491

**Comment by andreaswolf** [comment]
@cweiske Uh, I did not remember that :D Would have saved me quite a bit of time recently…

**Comment by evert** [comment]
uhoh :S so it looks like that fix never got ported to sabre/http when that split happened. Sorry about that

**Comment by evert** [comment]
Closed in favor of https://github.com/fruux/sabre-http/issues/47

**Comment by evert** [comment]
Hey guys, this is now fixed in sabre/http (tagged as 3.0.5) so you should be able to update this. If you could let me know it's not working for you, I will also prepare a full sabre/dav release.

**Comment by cweiske** [comment]
The new sabre-http version makes it work for me.
